### ISSUE TITLE
database-relation-broken hook fails when mysql primary unit is removed after relating with application charm

### ISSUE THREAD

**Issue by averma-canonical** [canonical/mysql-operator#32]
### Bug Description

In a charmed mysql cluster (after relating to the test application charm), if primary unit is removed, database-relation-broken hook gets triggered and fails with the following exception.

unit-mysql-1: 11:52:13 ERROR unit.mysql/1.juju-log database:2: Uncaught exception while in charm code:
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-mysql-1/charm/src/mysqlsh_helpers.py", line 172, in _run_mysqlsh_script
    return subprocess.check_output(command, stderr=subprocess.PIPE).decode("utf-8")
  File "/usr/lib/python3.8/subprocess.py", line 415, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/snap/bin/mysqlsh', '--no-wizard', '--python', '-f', '/root/snap/mysql-shell/common/tmpirx2nv08']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-mysql-1/charm/lib/charms/mysql/v0/mysql.py", line 858, in get_cluster_primary_address
    output = self._run_mysqlsh_script("\n".join(get_cluster_primary_commands))
  File "/var/lib/juju/agents/unit-mysql-1/charm/src/mysqlsh_helpers.py", line 174, in _run_mysqlsh_script
    raise MySQLClientError(e.stderr)
charms.mysql.v0.mysql.MySQLClientError: b'Cannot set LC_ALL to locale en_US.UTF-8: No such file or directory\nTraceback (most recent call last):\n  File "<string>", line 2, in <module>\nRuntimeError: Dba.get_cluster: This function is not available through a session to a standalone instance (metadata exists, instance does not belong to that metadata, and GR is not active)\n\n'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./src/charm.py", line 301, in <module>
    main(MySQLOperatorCharm)
  File "/var/lib/juju/agents/unit-mysql-1/charm/venv/ops/main.py", line 431, in main
    _emit_charm_event(charm, dispatcher.event_name)
  File "/var/lib/juju/agents/unit-mysql-1/charm/venv/ops/main.py", line 142, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-mysql-1/charm/venv/ops/framework.py", line 316, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-mysql-1/charm/venv/ops/framework.py", line 784, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-mysql-1/charm/venv/ops/framework.py", line 857, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-mysql-1/charm/src/relations/database.py", line 126, in _on_database_broken
    self.charm._mysql.delete_user_for_relation(relation_id)
  File "/var/lib/juju/agents/unit-mysql-1/charm/lib/charms/mysql/v0/mysql.py", line 441, in delete_user_for_relation
    primary_address = self.get_cluster_primary_address()
  File "/var/lib/juju/agents/unit-mysql-1/charm/lib/charms/mysql/v0/mysql.py", line 861, in get_cluster_primary_address
    raise MysqlGetClusterPrimaryAddressError(e.message)
charms.mysql.v0.mysql.MysqlGetClusterPrimaryAddressError: b'Cannot set LC_ALL to locale en_US.UTF-8: No such file or directory\nTraceback (most recent call last):\n  File "<string>", line 2, in <module>\nRuntimeError: Dba.get_cluster: This function is not available through a session to a standalone instance (metadata exists, instance does not belong to that metadata, and GR is not active)\n\n'
unit-mysql-1: 11:52:14 ERROR juju.worker.uniter.operation hook "database-relation-broken" (via hook dispatching script: dispatch) failed: exit status 1

### To Reproduce

1. juju deploy ./mysql_ubuntu-20.04-amd64.charm -n 2
2. juju deploy ./application_ubuntu-20.04-amd64.charm
3. juju relate application:database mysql:database
4. juju status                                                                              # to check the primary, assuming mysql/0 is the primary
5. juju remove-unit mysql/0                                                       # assuming mysql/0 is primary
6. juju status
7. juju debug-log

### Environment

- charm built from source code
- model on lxd controller (in a multipass sandbox)
- Ubuntu 20.04.4 LTS, focal

### Relevant log output

```shell
unit-mysql-1: 11:55:09 INFO juju.worker.uniter awaiting error resolution for "relation-broken" hook
unit-application-0: 11:56:17 INFO juju.worker.uniter.operation ran "update-status" hook (via hook dispatching script: dispatch)
unit-mysql-0: 11:58:22 INFO juju.worker.uniter.operation ran "update-status" hook (via hook dispatching script: dispatch)
unit-mysql-1: 11:59:21 INFO juju.worker.uniter awaiting error resolution for "relation-broken" hook
unit-mysql-1: 12:00:09 INFO juju.worker.uniter awaiting error resolution for "relation-broken" hook
unit-mysql-1: 12:00:09 WARNING unit.mysql/1.juju-log database:2: Failed to get cluster primary addresses
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-mysql-1/charm/src/mysqlsh_helpers.py", line 172, in _run_mysqlsh_script
    return subprocess.check_output(command, stderr=subprocess.PIPE).decode("utf-8")
  File "/usr/lib/python3.8/subprocess.py", line 415, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/snap/bin/mysqlsh', '--no-wizard', '--python', '-f', '/root/snap/mysql-shell/common/tmpu2zwb0fb']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-mysql-1/charm/lib/charms/mysql/v0/mysql.py", line 858, in get_cluster_primary_address
    output = self._run_mysqlsh_script("\n".join(get_cluster_primary_commands))
  File "/var/lib/juju/agents/unit-mysql-1/charm/src/mysqlsh_helpers.py", line 174, in _run_mysqlsh_script
    raise MySQLClientError(e.stderr)
charms.mysql.v0.mysql.MySQLClientError: b'Cannot set LC_ALL to locale en_US.UTF-8: No such file or directory\nTraceback (most recent call last):\n  File "<string>", line 2, in <module>\nRuntimeError: Dba.get_cluster: This function is not available through a session to a standalone instance (metadata exists, instance does not belong to that metadata, and GR is not active)\n\n'
unit-mysql-1: 12:00:09 ERROR unit.mysql/1.juju-log database:2: Uncaught exception while in charm code:
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-mysql-1/charm/src/mysqlsh_helpers.py", line 172, in _run_mysqlsh_script
    return subprocess.check_output(command, stderr=subprocess.PIPE).decode("utf-8")
  File "/usr/lib/python3.8/subprocess.py", line 415, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/snap/bin/mysqlsh', '--no-wizard', '--python', '-f', '/root/snap/mysql-shell/common/tmpu2zwb0fb']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-mysql-1/charm/lib/charms/mysql/v0/mysql.py", line 858, in get_cluster_primary_address
    output = self._run_mysqlsh_script("\n".join(get_cluster_primary_commands))
  File "/var/lib/juju/agents/unit-mysql-1/charm/src/mysqlsh_helpers.py", line 174, in _run_mysqlsh_script
    raise MySQLClientError(e.stderr)
charms.mysql.v0.mysql.MySQLClientError: b'Cannot set LC_ALL to locale en_US.UTF-8: No such file or directory\nTraceback (most recent call last):\n  File "<string>", line 2, in <module>\nRuntimeError: Dba.get_cluster: This function is not available through a session to a standalone instance (metadata exists, instance does not belong to that metadata, and GR is not active)\n\n'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./src/charm.py", line 301, in <module>
    main(MySQLOperatorCharm)
  File "/var/lib/juju/agents/unit-mysql-1/charm/venv/ops/main.py", line 431, in main
    _emit_charm_event(charm, dispatcher.event_name)
  File "/var/lib/juju/agents/unit-mysql-1/charm/venv/ops/main.py", line 142, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)  
  File "/var/lib/juju/agents/unit-mysql-1/charm/venv/ops/framework.py", line 316, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-mysql-1/charm/venv/ops/framework.py", line 784, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-mysql-1/charm/venv/ops/framework.py", line 857, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-mysql-1/charm/src/relations/database.py", line 126, in _on_database_broken
    self.charm._mysql.delete_user_for_relation(relation_id)
  File "/var/lib/juju/agents/unit-mysql-1/charm/lib/charms/mysql/v0/mysql.py", line 441, in delete_user_for_relation
    primary_address = self.get_cluster_primary_address()
  File "/var/lib/juju/agents/unit-mysql-1/charm/lib/charms/mysql/v0/mysql.py", line 861, in get_cluster_primary_address
    raise MysqlGetClusterPrimaryAddressError(e.message)
charms.mysql.v0.mysql.MysqlGetClusterPrimaryAddressError: b'Cannot set LC_ALL to locale en_US.UTF-8: No such file or directory\nTraceback (most recent call last):\n  File "<string>", line 2, in <module>\nRuntimeError: Dba.get_cluster: This function is not available through a session to a standalone instance (metadata exists, instance does not belong to that metadata, and GR is not active)\n\n'
```

**Comment by paulomach** [comment]
This is related to this [juju bug](https://bugs.launchpad.net/juju/+bug/1979811).

**Comment by paulomach** [comment]
Workaround implemented on PR #40
